.github/actions/cleanup/pe.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+func deleteAllPE(ctx context.Context, client mongodbatlas.PrivateEndpointsService, projectID string) error {
+	connections, err := getAllPrivateEndpoints(ctx, client, projectID)
+	if err != nil {
+		return fmt.Errorf("error getting private endpoints: %s", err)
+	}
+	err = deletePrivateEndpointsFromAtlas(ctx, client, projectID, connections)
+	if err != nil {
+		return fmt.Errorf("error deleting private endpoints: %s", err)
+	}
+	return nil
+}
+
+func deletePrivateEndpointsFromAtlas(ctx context.Context, client mongodbatlas.PrivateEndpointsService, projectID string, listsToRemove []mongodbatlas.PrivateEndpointConnection) error {
+	for _, peService := range listsToRemove {
+		if peService.Status == "DELETING" {
+			continue
+		}
+
+		if interfaceEndpointID(peService) != "" {
+			if _, err := client.DeleteOnePrivateEndpoint(ctx, projectID, peService.ProviderName, peService.ID, interfaceEndpointID(peService)); err != nil {
+				return fmt.Errorf("error deleting private endpoint interface: %s", err)
+			}
+
+			continue
+		}
+
+		if _, err := client.Delete(ctx, projectID, peService.ProviderName, peService.ID); err != nil {
+			return fmt.Errorf("error deleting private endpoint service: %s", err)
+		}
+	}
+	return nil
+}
+
+func interfaceEndpointID(connection mongodbatlas.PrivateEndpointConnection) string {
+
+	if len(connection.InterfaceEndpoints) != 0 {
+		return connection.InterfaceEndpoints[0]
+	}
+
+	if len(connection.PrivateEndpoints) != 0 {
+		return connection.PrivateEndpoints[0]
+	}
+
+	if len(connection.EndpointGroupNames) != 0 {
+		return connection.EndpointGroupNames[0]
+	}
+
+	return ""
+}
+
+func getAllPrivateEndpoints(ctx context.Context, client mongodbatlas.PrivateEndpointsService, projectID string) ([]mongodbatlas.PrivateEndpointConnection, error) {
```

Comment:
Not sure but I think some of these function can be taken from the operator. I think it's ok to use operator code as a dependency. Not an issue though since you've already coded it up :)

Reply:
Operator functions are non-exportable. And most of them are not suitable for the cleaner purpose.

Reply:
I think it would make sense if those functions were on their own module, making the internal behavior of the controller exported seems not a good idea.